controllers/navigationRoutes.js

```diff
@@ -52,28 +52,53 @@ router.get("/recipe/:id/edit", async (req, res) => {
   helper.SafeRequest(res, async (res) => {
 
     let recipe = {};
-  
+
     recipe.id = req.params.id
-  
-    if(!isNaN(recipe.id) && recipe.id != -1) {
+
+    if (!isNaN(recipe.id) && recipe.id != 0) {
       recipe = await getRecipeViewModel(recipe.id, req);
     } else {
       recipe.activeTimeUOM = "min";
       recipe.totalTimeUOM = "h";
     };
 
     recipe.timeUOMs = units.GetTimeUOMs().map(x => {
-      return {value: x.abbr, display: x.singular}
+      return { value: x.abbr, display: x.singular }
     })
-  
+
     res.render('recipe-edit', recipe);
 
   })
 })
 
 // GET route for dashboard page (user profile/account)
-router.get('/dashboard', (req, res) => {
-  res.render('dashboard')
+router.get('/dashboard', async (req, res) => {
+  helper.SafeRequest(res, async (res) => {
+    const user = await User.findOne({
+      where: {
+        id: req.session.userID
+      },
+      attributes: ['userName']
+    })
+    const userFavorites = await UserRecipeFavorite.findAll({
+      where: {
+        userID: req.session.userID
+      },
+      include: {
+        model: Recipe
+      }
+    })
+    const userRecipeFav = userFavorites.map(obj => obj.get().Recipe.get())
+
+    const userRecipes = await Recipe.findAll({
+      where: {
+        userID: req.session.userID
+      },
+    })
+
```

Comment:
looks like we have some duplicate code issues with this function being in unsecured and secured.